### PR TITLE
SearchBox: fluent update

### PR DESCRIPTION
**Pull request by maxwellred**
#### Pull request checklist

- [ ] Addresses an existing issue: #7008 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This updates the fluent theme-package changing some colors and hover states of the SearchBox component.

#### Changes

Updated clearButton background and icon color on rest and hover states
Screen Shot 2019-06-25 at 10 51 21 AM

Removed border radius on underlined variant
Screen Shot 2019-06-25 at 10 51 07 AM

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9475)

### CODE REVIEW

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
 
   return {
     root: [
       {
         borderRadius: effects.roundedCorner2,
-        borderColor: palette.neutralSecondaryAlt
+        borderColor: palette.neutralTertiary
+      },
+      underlined && [
+        'is-underlined',
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: No need for this class to be added in here as it's already in the default styles. Only provide the things you want to override.

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
+        {
+          borderRadius: '0px'
+        }
+      ]
+    ],
+    iconContainer: [
+      'ms-SearchBox-iconContainer',
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: Same here, remove the string classname

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 3]
```diff
@@ -1,14 +1,44 @@
...
+      {
+        color: palette.themeDarkAlt
+      }
+    ],
+    clearButton: [
+      'ms-SearchBox-clearButton',
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: same here. remove classname

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 4]
```diff
@@ -1,14 +1,44 @@
...
+      {
+        selectors: {
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: this entire selector feels wrong... If I understand the idea, you wanted to style the clearButton for its hover state, so why not having the hover selector nest the rest things. Let me know and I can help.

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 5]
```diff
@@ -1,14 +1,43 @@
...
   const { effects, palette } = theme;
 
   return {
     root: [
       {
         borderRadius: effects.roundedCorner2,
-        borderColor: palette.neutralSecondaryAlt
+        borderColor: palette.neutralTertiary
+      },
+      underlined && [
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: no need to wrap this object with square brackets. This way you make the life easier for merge-styles parsing phase 😄

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 6]
```diff
@@ -1,14 +1,43 @@
...
+        {
+          borderRadius: '0px'
+        }
+      ]
+    ],
+    iconContainer: [
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: same here...

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 7]
```diff
@@ -1,14 +1,43 @@
...
+      {
+        color: palette.themeDarkAlt
+      }
+    ],
+    clearButton: [
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: same here...

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 8]
```diff
@@ -1,15 +1,38 @@
...
+    },
+    clearButton: {
+      selectors: {
+        '&:hover': {
+          selectors: {
+            '.ms-Button': {
+              backgroundColor: 'transparent'
+            },
+            '.ms-Button-icon': {
+              color: palette.black
```
Comment (Vitalius1) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: @betrue-final-final are we using `black` for hover state? could this be a toolkit bug? I don't remember any hover states of any other component to be black.

Comment (maxwellred) [reply packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: I have spoken with Ben and changed it to neutralPrimary. Thanks @Vitalius1 !

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 9]
```diff
@@ -1,15 +1,38 @@
...
+        borderRadius: '0px'
       }
-    ]
+    ],
+    iconContainer: {
+      color: palette.themeDarkAlt
+    },
+    clearButton: {
+      selectors: {
+        '&:hover': {
```
Comment (ecraig12345) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: For speed of parsing, I wonder if these should be flattened:
```
selectors: {
  '&:hover .ms-Button': {...},
  '&:hover .ms-Button-icon': {...}
}
}
```

**Comment by maxwellred** [comment]
Closing PR and opening new branch in attempt to resolve unusual screener errors.

**Pull request closed without merging**